GUIDELINES.md

```diff
@@ -0,0 +1,27 @@
+# OpenTelemetry JS Code Contribution Guide
+
+This document outlines the essential guidelines for contributing code to the OpenTelemetry JS repository. These guidelines are designed to ensure consistency, stability, and the highest quality of code across the project.
+
+## Dependencies
+
+This section refers to `"dependencies"` and `"devDependencies"` entries in `package.json` file.
+It's important to note that not all libraries follow the `semver` convention, and those who do, might occasionally introduce breaking changes due to human errors (like putting breaking API changes in a patch version).
```

Comment:
Some rewording, added a sentence that allows us some wiggle-room in case we need to work around this. :slightly_smiling_face: 

```suggestion
> [!IMPORTANT]
> Not all libraries follow [Semantic Versioning](https://semver.org/). Even those who do might occasionally introduce breaking changes due to human errors. Exceptions to the guidelines in this document MAY be granted by Approvers or Maintainers to work around this.
```